tests/Umbraco.Tests.Integration/Umbraco.Examine.Lucene/UmbracoExamine/BackOfficeExamineSearcherTests.cs

```diff
@@ -0,0 +1,661 @@
+    {
+        long totalFound = long.MinValue;
+        return BackOfficeExamineSearcher.Search(query, UmbracoEntityTypes.Document,
+            pageSize, pageIndex, out totalFound, ignoreUserStartNodes: true);
+    }
+
+    private async Task SetupUserIdentity(string userId)
+    {
+        var identity =
+            await BackOfficeUserStore.FindByIdAsync(userId, CancellationToken.None);
+        await BackOfficeSignInManager.SignInAsync(identity, false);
+    }
+
+    private PublishResult CreateDefaultPublishedContent(string contentName)
+    {
+        var contentType = new ContentTypeBuilder()
+            .WithId(0)
+            .Build();
+        ContentTypeService.Save(contentType);
+
+        var content = new ContentBuilder()
+            .WithId(0)
+            .WithName(contentName)
+            .WithContentType(contentType)
+            .Build();
+        var createdContent = ContentService.SaveAndPublish(content);
+
+        // It seems like we need this timeout for the moment, otherwise our indexes would not be built in time and the test could fail.
+        Thread.Sleep(1000);
+
+        return createdContent;
+    }
+
+    private PublishResult CreateDefaultPublishedContentWithTwoLanguages(string englishNodeName, string danishNodeName)
+    {
+        string usIso = "en-US";
+        string dkIso = "da";
+
+        var langDa = new LanguageBuilder()
+            .WithCultureInfo(dkIso)
+            .Build();
+        LocalizationService.Save(langDa);
+
+        var contentType = new ContentTypeBuilder()
+            .WithId(0)
+            .WithContentVariation(ContentVariation.Culture)
+            .Build();
+        ContentTypeService.Save(contentType);
+
+        var content = new ContentBuilder()
+            .WithId(0)
+            .WithCultureName(usIso, englishNodeName)
+            .WithCultureName(dkIso, danishNodeName)
+            .WithContentType(contentType)
+            .Build();
+        var createdContent = ContentService.SaveAndPublish(content);
+        Thread.Sleep(1000);
+
+        return createdContent;
+    }
+
+    [Test]
+    public void Search_Published_Content_With_Empty_Query()
+    {
+        SetupUserIdentity(Constants.Security.SuperUserIdAsString);
+
+        // Arrange
+        var contentName = "TestContent";
+        CreateDefaultPublishedContent(contentName);
+
+        string query = string.Empty;
+
+        // Act
+        IEnumerable<ISearchResult> actual = BackOfficeExamineSearch(query);
+
+        // Assert
+        Assert.AreEqual(0, actual.Count());
+    }
+
+    [Test]
+    public void Search_Published_Content_With_Query_By_Content_Name()
+    {
+        // Arrange
+        SetupUserIdentity(Constants.Security.SuperUserIdAsString);
+
+        string contentName = "TestContent";
+        CreateDefaultPublishedContent(contentName);
+
+        string query = contentName;
+
+        // Act
+        IEnumerable<ISearchResult> actual = BackOfficeExamineSearch(query);
```

Comment:
Multiple enumerations here 😄as a rule `IEnumerable` can only be run through once, so to get around this you need to call `.ToArray()` on it and then make the variable an array (or ToList(), and list variable)